django/db/models/base.py

```diff
@@ -971,8 +971,8 @@ def _get_next_or_previous_by_FIELD(self, field, is_next, **kwargs):
         op = 'gt' if is_next else 'lt'
         order = '' if is_next else '-'
         param = getattr(self, field.attname)
-        q = Q(**{'%s__%s' % (field.name, op): param})
-        q = q | Q(**{field.name: param, 'pk__%s' % op: self.pk})
+        q = Q((field.name, param), ('pk__%s' % op, self.pk), _connector=Q.AND)
+        q = Q(q, ('%s__%s' % (field.name, op), param), _connector=Q.OR)
         qs = self.__class__._default_manager.using(self._state.db).filter(**kwargs).filter(q).order_by(
```

Comment:
Neither of these is particularly beautiful to look at, but I've opted to do it so that it can be highlighted and discussed and optionally leave it as it was.